python/whylogs/core/constraints/factories/distribution_metrics.py

```diff
@@ -135,20 +164,18 @@ def quantile_between_range(column_name: str, quantile: float, lower: float, uppe
         Upper bound of the value range
     skip_missing: bool
         If skip_missing is True, missing distribution metrics will make the check pass.
-        If False, the check will fail on missing metrics
+        If False, the check will fail on missing metrics, such as on an empty dataset
     """
 
     def quantile_in_range(metric):
-        if not metric.kll.value.is_empty():
-            quantile_value = metric.kll.value.get_quantile(quantile)
-            result: bool = lower <= quantile_value <= upper
-            return result
-        else:
-            return True if skip_missing else False
+        quantile_value = metric.kll.value.get_quantile(quantile)
```

Comment:
The `Predicate`s grab values to use in expressions from the metric summary, so I don't think they could handle this case. You're limited to the quantiles hard-coded in the summary.